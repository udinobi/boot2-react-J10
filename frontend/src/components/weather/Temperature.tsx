import React from "react"

import styled from "styled-components"

import { Option } from "ts-option"

import { WeatherData } from "../../store/weather/types"

export interface Data {
    weatherData: Option<WeatherData>
}

const Celsius = styled.sup`
    color: #212121;
    font-size: 28px;
`

const Degrees = styled.span`
    color: #212121;
    font-family: "Roboto", "sans-serif";
    font-size: 64px;
    padding-right: 6px;
`

const Icon = styled.i`
    font-size: 64px;
    padding-right: 22px;
`

const dayIconColor = (temp: number) => {
    /* tslint:disable:curly */
    if (temp <  5) return "day-under-5-icon-color"
    if (temp < 12) return "day-under-12-icon-color"
    if (temp < 22) return "day-under-22-icon-color"
    if (temp < 32) return "day-under-32-icon-color"
    return "day-from-32-icon-color"
}

const iconClasses = (weatherData: WeatherData) => {
    const now = Math.floor(Date.now() / 1000)
    return now >= weatherData.sys.sunrise && now <= weatherData.sys.sunset
        ? `${dayIconColor(weatherData.main.temp)} wi wi-owm-day`
        : "night-icon-color wi wi-owm-night"
}

const icon = (weatherData: WeatherData) =>
    weatherData.weather.length > 0
        ? <Icon className={`${iconClasses(weatherData)}-${weatherData.weather[0].id}`} />
        : <div style={{ height: 64, paddingRight: 22, width: 64 }} />

export default (props: Data) =>
    props.weatherData.map(weatherData => (
        <div>
            {icon(weatherData)}
            <Degrees>{Math.round(weatherData.main.temp)}</Degrees>
            <Celsius>Â°C</Celsius>
        </div>
    ))
    .getOrElse(<div />)
