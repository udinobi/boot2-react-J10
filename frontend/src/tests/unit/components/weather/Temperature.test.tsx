import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import React from "react"

import { none } from "ts-option"

import Temperature, { Data } from "../../../../components/weather/Temperature"

describe("--> Snapshot for Components > weather > 'Temperature'", () => {
    let props: Data

    beforeEach(() => {
        props = {
            weatherData: none
        }
    })

    it("matches", () => {
        const wrapper = shallow(<Temperature {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
