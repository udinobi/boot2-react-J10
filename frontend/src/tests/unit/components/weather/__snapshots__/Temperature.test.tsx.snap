// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > weather > 'Temperature' matches 1`] = `<div />`;
